### PR TITLE
[pickers] Add key in the `localeText`

### PR DESCRIPTION
**Pull request by alexfauquette**
This is about the pen/clock in toolbars. There is maybe other texts I missed

Part of #4446

image
image

### CODE REVIEW

File: packages/x-date-pickers/src/internals/components/PickersToolbar.tsx [hunk 1]
```diff
@@ -129,7 +121,10 @@ export const PickersToolbar = React.forwardRef(function PickersToolbar<TDate, TV
           ownerState={ownerState}
           color="inherit"
           data-mui-test="toggle-mobile-keyboard-view"
-          aria-label={getMobileKeyboardInputViewButtonText(isMobileKeyboardViewOpen, viewType)}
+          aria-label={localeText.toolbarInputTypeSwitchingButtonAriaLabel(
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/internals/components/PickersToolbar.tsx:124]:
```suggestion
          aria-label={localeText.toolbarInputToViewSwitchButtonAriaLabel(
```

Right now I don't understand what the button is doing just with the locale key.
Mine is not great either.

Do we agree that this button is used to toggle the input inside the modal in mobile ?

Comment (alexfauquette) [reply packages/x-date-pickers/src/internals/components/PickersToolbar.tsx:124]: Yes, except that it is not necessary on mobile. 
With `showToolbar` prop you can let the toolbar (and so the button) appears on desktop too

Comment (alexfauquette) [reply packages/x-date-pickers/src/internals/components/PickersToolbar.tsx]: I tried with `toolbarInterfaceModifierButtonAriaLabel` but the real problem is this button is rarely used

Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/components/PickersToolbar.tsx]: @m4theushw if you have an idea
All mine are crappy :laughing:

Comment (alexfauquette) [reply packages/x-date-pickers/src/internals/components/PickersToolbar.tsx]: @LukasTy Maybe you will have an idea about how to name this localisation key

File: packages/x-date-pickers/src/internals/components/PickersToolbar.tsx [hunk 2]
```diff
@@ -129,7 +121,10 @@ export const PickersToolbar = React.forwardRef(function PickersToolbar<TDate, TV
           ownerState={ownerState}
           color="inherit"
           data-mui-test="toggle-mobile-keyboard-view"
-          aria-label={getMobileKeyboardInputViewButtonText(isMobileKeyboardViewOpen, viewType)}
+          aria-label={localeText.toolbarInterfaceModifierButtonAriaLabel(
```
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/components/PickersToolbar.tsx:124]: Maybe naming it `inputModeToggleButtonAriaLabel` or `inputViewToggleButtonAriaLabel` would make the field name more clear to the consumer?

At the same time, renaming `isMobileKeyboardViewOpen` to `isKeyboardInputOpen` everywhere would also make sense, but, can of course be left for a future refactor. 😄

File: packages/x-date-pickers/src/locales/itIT.ts [hunk 3]
```diff
@@ -33,6 +32,12 @@ const itITPickers: PickersLocaleText<any> = {
   okButtonLabel: 'OK',
   todayButtonLabel: 'Oggi',
 
+  // Toolbar titles
+  datePickerDefaultToolbarTitle: '',
+  dateTimePickerDefaultToolbarTitle: '',
+  timePickerDefaultToolbarTitle: '',
+  dateRangePickerDefaultToolbarTitle: '',
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply packages/x-date-pickers/src/locales/itIT.ts:39]: Yes, that strange because those lines should be added by a script

**Review by LukasTy — APPROVED** [review]

File: packages/x-date-pickers/src/internals/models/props/baseToolbarProps.ts [hunk 4]
```diff
@@ -11,7 +11,6 @@ export interface BaseToolbarProps<TDate, TValue>
     Omit<PickerStatePickerProps<TValue>, 'onDateChange'> {
   ampmInClock?: boolean;
   dateRangeIcon?: React.ReactNode;
-  getMobileKeyboardInputViewButtonText?: () => string;
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply packages/x-date-pickers/src/internals/models/props/baseToolbarProps.ts:14]: Yes, that's feasible

**Pull request merged**